modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationSpi.java

```diff
@@ -2323,6 +2318,14 @@ private void dumpInfo(StringBuilder sb, UUID dstNodeId) {
             throw new IgniteSpiException("Failed to initialize TCP server: " + locHost, e);
         }
 
+        EnvironmentType envType = ignite.configuration().getEnvironmentType();
+
+        if (usePairedConnections) {
+            if (envType == EnvironmentType.VIRTUALIZED)
+                throw new IgniteSpiException("Node using paired connections " +
+                    "is not allowed to start in virtualized environment.");
```

Comment:
We need to suggest which configuration properties a user has to change in order to resolve this error